apps/hash-graph/lib/graph/src/store/postgres/ontology/entity_type.rs

```diff
@@ -25,197 +25,183 @@ impl<C: AsClient> PostgresStore<C> {
                     }
                 }
 
-                if let Some(inherits_from_type_ref_urls) = inherits_from_type_ref_urls {
-                    for inherits_from_type_ref_url in inherits_from_type_ref_urls {
-                        let inherits_from_type_vertex_id =
-                            EntityTypeVertexId::from(inherits_from_type_ref_url);
+                if graph_resolve_depths.inherits_from.outgoing > 0 {
+                    for referenced_entity_type in <Self as Read<EntityTypeWithMetadata>>::read(
+                        self,
+                        &Filter::<EntityTypeWithMetadata>::for_ontology_edge_by_entity_type_vertex_id(
+                            &entity_type_vertex_id,
+                            OntologyEdgeKind::InheritsFrom,
+                            true,
+                        ),
+                        &temporal_axes,
+                    )
+                        .await?
+                    {
+                        let referenced_entity_type_vertex_id = referenced_entity_type.vertex_id(time_axis);
 
                         subgraph.insert_edge(
-                            &entity_type_id,
+                            &entity_type_vertex_id,
                             OntologyEdgeKind::InheritsFrom,
-                            inherits_from_type_vertex_id.clone(),
+                            referenced_entity_type_vertex_id.clone(),
                         );
 
-                        queue.push((
-                            inherits_from_type_vertex_id,
+                        subgraph.insert_vertex(&referenced_entity_type_vertex_id, referenced_entity_type);
+
+                        entity_type_queue.push((
+                            referenced_entity_type_vertex_id,
                             GraphResolveDepths {
                                 inherits_from: OutgoingEdgeResolveDepth {
-                                    outgoing: current_resolve_depths.inherits_from.outgoing - 1,
-                                    ..current_resolve_depths.inherits_from
+                                    outgoing: graph_resolve_depths.inherits_from.outgoing
+                                        - 1,
+                                    ..graph_resolve_depths.inherits_from
+                                },
+                                ..graph_resolve_depths
+                            },
+                            temporal_axes.clone()
+                        ));
+                    }
+                }
+
+                if graph_resolve_depths.constrains_links_on.outgoing > 0 {
+                    for referenced_entity_type in <Self as Read<EntityTypeWithMetadata>>::read(
+                        self,
+                        &Filter::<EntityTypeWithMetadata>::for_ontology_edge_by_entity_type_vertex_id(
+                            &entity_type_vertex_id,
+                            OntologyEdgeKind::ConstrainsLinksOn,
+                            true,
```

Comment:
Why is this reversed?

Reply:
Considering these are all standalone functions (effectively aliases for specific filters) do we want to define them the other way around so they're more intuitive and more like how we'd think of writing a query?

"Get me all outgoing links of this entity"

Because now I'm going to look at this code:
https://github.com/hashintel/hash/blob/a30bc9e4ea3b8b83dc0b5cdefa9d7c16256391da/apps/hash-graph/lib/graph/src/store/postgres/knowledge/entity.rs#L123-L130

And [Fix traversal direction](https://github.com/hashintel/hash/pull/2223/commits/de548a6431d89fd2147ba9a6e9456f7326db7251) means I'm going to be even more confused

As I'm going to look at the code for correctness and think, well
an `outgoing_link_entity` relationship is equivalent to `(kind=left_entity, reversed=true)`

And I'm going to look at your filter and see `left_entity, false` and think it's a bug

Reply:
Hmm okay, thinking more about how the filters work, I guess swapping it around would be confusing and inconsistent with other kinds of filters..

Reply:
The current end of the stack still have those, but I expect to completely remove these and only have `Filter::for_edge_by_vertex_id`

Reply:
> The current end of the stack still have those, but I expect to completely remove these and only have Filter::for_edge_by_vertex_id

Won't you need to write it the other way around at that point? If you're completely abstracting over just the edge by the vertex id, will you know the type of the destination you're querying?

I just find it very unintuitive that we're abstracting it to the point that we're doing every query in reverse. We're traversing a graph and we're trying to get all edges going from a thing, and instead we're querying for everything which has an edge _to_ it from the thing.. That just seems backwards